### ISSUE TITLE
Gif partially lose transparency / misalign, not on all frames

### ISSUE THREAD

**Issue by Nitaym** [python-pillow/Pillow#2239]
I'm building a script to extract a frame from a gif, and encountered a bug (I think).
The script is working properly with several gifs I've tried it on, but this specific gif I'm getting weird misalignments and transparency issues.

This is the original gif
![loading](https://cloud.githubusercontent.com/assets/19125/20505815/28a47b1c-b058-11e6-9e8f-aab9ec08009d.gif)

And this is the output from frame 10:
![sprite](https://cloud.githubusercontent.com/assets/19125/20505816/2a525ccc-b058-11e6-8cc1-47a118ec6ef3.png)

This happens on almost every frame.

Here's the python code to extract the frame:
```python
with open('loading.gif', "rb") as f:
    gif_data = bytearray(f.read())

try:
    # gif = Image.open(infile)
    gif = Image.open(io.BytesIO(gif_data))
    width, height = gif.size
except IOError as e:
    print "Cant load", gif_data
    sys.exit(1)

mypalette = gif.getpalette()

sprite = Image.new("RGBA", (width, height))
for i in range(10):
    gif.seek(gif.tell() + 1)

gif.putpalette(mypalette)
sprite.paste(gif, (0, 0, width, height))

sprite_data = io.BytesIO()
sprite.save(sprite_data , format='PNG')
sprite_data = sprite_data.getvalue()

with open('sprite.png', 'wb') as f:
    # png_base64 = (json.load(io.BytesIO(converted)))['file']
    # png = base64.b64decode(png_base64)
    f.write(sprite_data)
    print "File written"
```

**Comment by wiredfool** [comment]
That one is weird. I've never seen imagemagick's display have a fully transparent window before. 

It's going to take some digging, but I'm suspecting that some of the trouble comes from having the first frame not actually cover the entire size of the gif.

**Comment by radarhere** [comment]
Just to provide simpler code to reproduce the issue -
```
from PIL import Image
im = Image.open('test.gif')
im.seek(10)
im.save("out.png")
```

**Comment by hantaogo** [comment]
我有许多图片遇到了这个BUG，希望能早点解决
I meet this BUG too, I hope this bug could be closed soon!

**Comment by radarhere** [comment]
#5333 fixed the blue area.
It did create a new problem, extending the circle arcs, but #5391 fixes that.
The black and white background is still there instead of transparency, but we're making progress.

**Comment by radarhere** [comment]
This has been resolved by #5557
